### DISCUSSION TITLE
previous changes to function patchAttributes(from2, to) cause bugs

### DISCUSSION THREAD

**Discussion by schoolshark** [livewire/livewire#8975] [7. Bugs (archived)] [answered]
### Livewire version

3.5.12

### Laravel version

11

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

Hello :-)

With the latest Livewire update, the behavior for marking HTML table cells has changed. I use style attributes to mark table cells, which preserves the original class attributes and allows the original color to be restored when deselected. This approach had been working very well for me, even persisting through re-renderings.  However, with Livewire version 3.5.12, my code suddenly stopped working.  

1. Step to reproduce:   set a style attribute
e.g. myExampleTableCell.style.backgroundColor = someColor;
2. Step to reproduce: wait for next render() 

I did a diff of the Livewire JavaScript assets and found that the following two lines has been removed:

// now missing —> caused my bug 
if (name === "style")
      continue;

After re-insert this lines everything is fine, as before.

// from livewire.js:
function patchAttributes(from2, to) {
  if (from2._x_transitioning)
    return;
  if (from2._x_isShown && !to._x_isShown) {
    return;
  }
  if (!from2._x_isShown && to._x_isShown) {
    return;
  }
  let domAttributes = Array.from(from2.attributes);
  let toAttributes = Array.from(to.attributes);
  for (let i = domAttributes.length - 1; i >= 0; i--) {
    let name = domAttributes[i].name;
     if (name === "style")
      continue;
    if (!to.hasAttribute(name)) {
      from2.removeAttribute(name);
    }
  }
  for (let i = toAttributes.length - 1; i >= 0; i--) {
    let name = toAttributes[i].name;
    let value = toAttributes[i].value;
    if (name === "style")
      continue;
    if (from2.getAttribute(name) !== value) {
      from2.setAttribute(name, value);
    }
  }
}

### Repository or Wirebox URL

https://github.com/livewire/livewire

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment, marked as answer]
For anyone else coming across this issue, see my comment on this other discussion https://github.com/livewire/livewire/discussions/9065#discussioncomment-11456086

@schoolshark please don't open two discussions for the same issue in the future, thanks 🙂
